### ISSUE TITLE
r.sample() changes

### ISSUE THREAD

**Issue by wojons** [rethinkdb/rethinkdb#2289]
The r.sample needs 2 to 3 premature to be really useful. It may even make it easier to program.

limit, sample size, sample rate.

For example. r.sample(100,5,2) could mean. for every 5 documents randomly pick to. but never take more than 100.

I think it may be better to put the total take away outside of the sample so it would be handled like.

r.sample(5, 2).limit(100) this way the it would still mean the same thing but would be using normal reql language.

**Comment by mbroadst** [comment]
I'll second that +:100:

**Comment by mholt** [comment]
Yes, this would be fantastic! (I know it's been like 3 years but I just had to add my support of this motion.) If anyone has time to work on it, I think a lot of us would appreciate it. :smile:
